### PR TITLE
fix: fixing issue with missing dependency

### PR DESCRIPTION
**Pull request by johnwalz97**
## Internal Notes for Reviewers

Adding logic to handle tests that have missing dependencies in a standardized way. This will let users list and load tests without having the dependencies those tests require but it will notify them of this so they aren't missing out on any tests.

image

### CODE REVIEW

**Comment by cachafla** [comment]
Oops I forgot to merge this other PR 😬 https://github.com/validmind/developer-framework/pull/205. I won't do it anymore since you added this @johnwalz97

File: pyproject.toml [hunk 1]
```diff
@@ -13,12 +13,13 @@ readme = "README.pypi.md"
 version = "2.5.19"
 
 [tool.poetry.dependencies]
+python = ">=3.8.1,<3.12"
```
Comment (cachafla) [inline pyproject.toml:16]: This will keep moving on some of us' editor formatters that have auto-sort dependencies enabled 😅

**Review by cachafla — APPROVED** [review]
Awesome 👏. Could you target this PR to merge into `prod` instead?

File: pyproject.toml (hunk 1 above)
Comment (johnwalz97) [reply pyproject.toml:16]: lol I've been meaning to add something to the readme to install the poetry auto-sort plugin so we all have the same auto-sorter 😂

**Comment by johnwalz97** [comment]
> Oops I forgot to merge this other PR 😬 #205. I won't do it anymore since you added this @johnwalz97

all good... either technically works to fix the issue but i figured this would be a more standardized approach that we can start adopting 😅

**Pull request merged**